reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/delegations/DelegationsManagement.scala

```diff
@@ -33,6 +43,16 @@ class DelegationsManagement()(implicit val db: Database, val ec: ExecutionContex
     targetsRole.delegations.flatMap(_.roles.find(_.name == delegatedRoleName)).getOrElse(throw Errors.DelegationNotDefined)
   }
 
+  def findDelegations(targets: SignedRole[TargetsRole]): Future[Map[MetaPath, MetaItem]] = {
```

Comment:
It also reads the actual delegated roles from the `delegationsRepo`. Maybe the issue is that the delegated role only have their own type in the DB and is not a TufRole? Which means we can't use the existing SignedRole[T:TufRole].asMetaRole. (Using it as a `Targets` `TufRole` doesn't work since asMetaRole uses the type of the TufRole as a name, which would need to be a variable for a delegated role.)